include/onnxruntime/core/optimizer/rewrite_rule.h

```diff
@@ -35,6 +35,17 @@ If the list of op types is left empty, that rule will be triggered for every op
 */
 class RewriteRule {
  public:
+  /**
+  @class RewriteRuleEffect
+
+  Class used to indicate the effect of rule application on a graph's node.
+  */
+  enum class RewriteRuleEffect : uint8_t {
+    kUnmodifiedGraph,    // The graph has not been modified by the rewrite rule.
+    kModifiedGraph,      // The graph was modified (some nodes were updated or removed).
+    kRemovedCurrentNode  // The graph was modified and the node on which the rewrite rule was triggered is removed.
+  };
```

Comment:
It feels like there are two different aspects here. One is whether the graph was modified. One is how it was modified. Should we maybe do this as flags (one bit for denoting whether the graph was modified, other bits for specifying the type/s of modification)?

Or alternatively make the documentation a little more explicit. e.g. kModifiedGraph could say  "The graph was modified (some nodes, other than the one the rewrite rule was triggered on, were updated or removed)".

Reply:
I can create a class and keep a modified_graph boolean and an enum that indicates the modification reason (at the moment it's only two reasons, but I want to keep it extensible).
But I can instead have the following values in the enum:
kNone (instead of kUnmodified as you suggested), kUpdatedCurrentNode, kRemovedCurrentNode, kRestOfGraphModified.
I think the latter is simpler and cleaner.

Reply:
Done with latest commit.